core/contextmenu.js

```diff
@@ -404,4 +404,99 @@ Blockly.ContextMenu.wsExpandOption = function(hasCollapsedBlocks, topBlocks) {
   };
 };
 
+/**
+ * Make a context menu option for deleting the current workspace comment.
+ * @param {!Blockly.WorkspaceCommentSvg} comment The workspace comment where the
+ *     right-click originated.
+ * @return {!Object} A menu option, containing text, enabled, and a callback.
+ * @package
+ */
+Blockly.ContextMenu.commentDeleteOption = function(comment) {
+  var deleteOption = {
+    text: Blockly.Msg.REMOVE_COMMENT,
+    enabled: true,
+    callback: function() {
+      Blockly.Events.setGroup(true);
+      comment.dispose(true, true);
+      Blockly.Events.setGroup(false);
+    }
+  };
+  return deleteOption;
+};
+
+/**
+ * Make a context menu option for duplicating the current workspace comment.
+ * @param {!Blockly.WorkspaceCommentSvg} comment The workspace comment where the
+ *     right-click originated.
+ * @return {!Object} A menu option, containing text, enabled, and a callback.
+ * @package
+ */
+Blockly.ContextMenu.commentDuplicateOption = function(comment) {
+  var duplicateOption = {
+    text: Blockly.Msg.DUPLICATE_COMMENT,
+    enabled: true,
+    callback: function() {
+      Blockly.duplicate_(comment);
+    }
+  };
+  return duplicateOption;
+};
+
+/**
+ * Make a context menu option for adding a comment on the workspace.
+ * @param {!Blockly.WorkspaceSvg} ws The workspace where the right-click
+ *     originated.
+ * @param {!Event} e The right-click mouse event.
+ * @return {!Object} A menu option, containing text, enabled, and a callback.
+ * @package
+ */
+Blockly.ContextMenu.workspaceCommentOption = function(ws, e) {
+  // Helper function to create and position a comment correctly based on the
+  // location of the mouse event.
+  var addWsComment = function() {
+    var comment = new Blockly.WorkspaceCommentSvg(
+        ws, Blockly.Msg.WORKSPACE_COMMENT_DEFAULT_TEXT,
+        Blockly.WorkspaceCommentSvg.DEFAULT_SIZE,
+        Blockly.WorkspaceCommentSvg.DEFAULT_SIZE);
```

Comment:
While playing around with the vertical_playground, I noticed that the size of the workspace comment is very small compared to the size of the blocks. I'm not sure what size we're aiming for, but it looks like it's easy enough to change! =]